Line1

BelousEV opened the pull request.

BelousEV committed:
Готово (без тестов)

BelousEV committed:
student

BelousEV committed:
Почти готово

dmitry-bizin commented on pom.xml:
```diff
@@ -27,6 +46,14 @@
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
```
зачем 2 раза подключать одну и ту же зависимость?)

dmitry-bizin commented on pom.xml:
```diff
@@ -17,6 +17,25 @@
     <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
```
эта зависимость не нужна, она есть в spring-boot-starter-test

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,76 @@
+public class FacultyController {
+
+
+    private FacultyServiceImpl facultyServiceImpl;
```
здесь нужно использовать интерфейс

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,76 @@
+
+    private FacultyServiceImpl facultyServiceImpl;
+
+    public FacultyController(FacultyServiceImpl facultyServiceImpl) {
```
здесь надо инжектить тоже по интерфейсу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,76 @@
+    @GetMapping("{id}") //http://localhost:8081/faculties/23
+
+    public ResponseEntity <Faculty> getFacultyInfo(@PathVariable long id) {
+        Faculty faculty = FacultyServiceImpl.findFaculty(id);
```
здесь надо обращаться к объекту, а не классу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,76 @@
+    public ResponseEntity <Faculty> getFacultyInfo(@PathVariable long id) {
+        Faculty faculty = FacultyServiceImpl.findFaculty(id);
+        if (faculty == null) {
+            ResponseEntity.notFound().build();
```
return

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,76 @@
+
+    @PostMapping // POST http://localhost:8081/faculties
+    public Faculty createFaculty(@RequestBody Faculty faculty) {
+        return FacultyServiceImpl.createFaculty(faculty);
```
здесь надо обращаться к объекту, а не классу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,76 @@
+    }
+    @PutMapping //http://localhost:8081/faculties
+    public ResponseEntity <Faculty> editFaculty(@RequestBody Faculty faculty){
+        Faculty foundFaculty = FacultyServiceImpl.editFaculty(faculty);
```
здесь надо обращаться к объекту, а не классу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,76 @@
+
+    public ResponseEntity deleteFaculty(@PathVariable long id) {
+
+        FacultyServiceImpl.deleteFaculty(id);
```
здесь надо обращаться к объекту, а не классу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,76 @@
+
+    @DeleteMapping ("{id}") //удаление по айди http://localhost:8081/faculties/23
+
+    public ResponseEntity deleteFaculty(@PathVariable long id) {
```
ResponseEntity\<?>

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,73 @@
+public class StudentController {
+
+
+    private final StudentServiceImpl studentService;
```
здесь нужно использовать интерфейс

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,73 @@
+
+    private final StudentServiceImpl studentService;
+
+    public StudentController(StudentServiceImpl studentService) {
```
здесь надо инжектить тоже по интерфейсу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,73 @@
+
+    @GetMapping("{id}") //GET http://localhost:8081/students/23
+    public ResponseEntity <Student> getStudentInfo(@PathVariable long id) {
+        Student student = StudentServiceImpl.findStudent(id);
```
здесь надо обращаться к объекту, а не классу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,73 @@
+    @PostMapping  // POST http://localhost:8081/students
+    public Student createStudent(@RequestBody Student student) {
+
+        return StudentServiceImpl.createStudent(student);
```
здесь надо обращаться к объекту, а не классу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,73 @@
+
+    @PutMapping  //PUT редактирование http://localhost:8081/students
+    public ResponseEntity <Student> editStudent (@RequestBody Student student){
+      Student foundStudent = StudentServiceImpl.editStudent(student);
```
здесь надо обращаться к объекту, а не классу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,73 @@
+    @DeleteMapping ("{id}") //DELETE http://localhost:8081/students/23
+
+    public ResponseEntity deleteStudent(@PathVariable long id) {
+        StudentServiceImpl.deleteStudent(id);
```
здесь надо обращаться к объекту, а не классу

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,73 @@
+
+    @DeleteMapping ("{id}") //DELETE http://localhost:8081/students/23
+
+    public ResponseEntity deleteStudent(@PathVariable long id) {
```
ResponseEntity\<?>

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/FacultyServiceImpl.java:
```diff
@@ -0,0 +1,54 @@
+    }
+    // Service
+
+
```
здесь нужно написать метод для поиска факультетов по цвету. Этот метод будет вызывать метод репозитория. В репозитории нужно написать метод для поиска по цвету в БД

dmitry-bizin replied:
по-прежнему актуально

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/StudentServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+    public Collection<Student> getAllStudent(){
+        return studentRepository.findAll();
+    }
+
```
здесь нужно написать метод для поиска студентов по возрасту. Этот метод будет вызывать метод репозитория. В репозитории нужно написать метод для поиска по возрасту в БД

dmitry-bizin replied:
по-прежнему актуально

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -1 +1,8 @@
+spring.datasource.username= student
+spring.datasource.password= chocolatefrog
+
+Hibernate ddl auto (create, create-drop, validate, update)
```
эту строку можно удалить

BelousEV committed:
Готово

BelousEV committed:
Готово

BelousEV committed:
Готово

BelousEV committed:
Merge remote-tracking branch 'origin/line1' into line1

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,73 @@
+    // Controller
+    @GetMapping
+    public ResponseEntity<Collection<Student>> findStudents(@RequestParam(required = false) Long age) {
+        if (age > 0) {
```
age может быть наллом, если required=false стоит, нужно бы проверить на налл

dmitry-bizin commented on src/main/java/ru/hogwarts/school/model/Faculty.java:
```diff
@@ -0,0 +1,70 @@
+@Entity
+public class Faculty {
+    @javax.persistence.Id
+    @Id
```
эта аннотация лишняя, должна быть только из пакета persistence

dmitry-bizin commented on src/main/java/ru/hogwarts/school/model/Student.java:
```diff
@@ -0,0 +1,74 @@
+public class Student {
+
+    @javax.persistence.Id
+    @Id
```
эта аннотация лишняя, должна быть только из пакета persistence

BelousEV replied:
Без данной аннотации (проставляется автоматически) все подчеркивает красным =/

dmitry-bizin replied:
не может быть)

dmitry-bizin replied:
давай скрин тогда)

BelousEV replied:
А не, все, разобралась )

BelousEV committed:
Готово

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/StudentServiceImpl.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public Student findStudent(Long id) {
+        return studentRepository.getOne(id);
```
findById

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/FacultyServiceImpl.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public  Faculty findFaculty(Long id) {
+        return facultyRepository.getReferenceById(id);
```
findById

BelousEV committed:
Готово

BelousEV committed:
Готово_2

BelousEV committed:
ДЗ 3.4

BelousEV committed:
ДЗ 3.4

BelousEV committed:
ДЗ 3.4 поправила

dmitry-bizin commented on scripts.sql:
```diff
@@ -0,0 +1,12 @@
+select * from student
+where age >20 and age <30;
```
можно использовать between

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,75 @@
+public class FacultyController {
+
+
+    private FacultyService facultyService;
```
final

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,75 @@
+        return ResponseEntity.ok(facultyService.getAll());//вероятно, лишнее о_О
+    }
+    @GetMapping
+    public ResponseEntity findByNameOrColorIgnoreCase(@RequestParam String name, String color) {
```
по заданию здесь один параметр - String nameOrColor. Если у тебя 2, то перед каждым надо ставить аннотацию

dmitry-bizin replied:
ResponseEntity надо использовать с дженерик-параметром

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,75 @@
+        return ResponseEntity.ok(Collections.emptyList());
+    }
+
+
```
нет метода получения студентов факультета

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    @GetMapping // для SQL БД (пока не фурычит)
+    public ResponseEntity findStudent(@RequestParam Long age_min, Long age_max) {
```
названия параметров в camelCase, также аннотации надо ставить перед каждым параметром

dmitry-bizin replied:
ResponseEntity надо использовать с дженерик-параметром

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,83 @@
+
+    @GetMapping // для SQL БД (пока не фурычит)
+    public ResponseEntity findStudent(@RequestParam Long age_min, Long age_max) {
+        if (age_max != null && age_min != null) {
```
параметры не могут быть налл, т.к. по умолчанию required=true

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -0,0 +1,75 @@
+    }
+    @GetMapping
+    public ResponseEntity findByNameOrColorIgnoreCase(@RequestParam String name, String color) {
+        if (name != null && !name.isBlank()) {
```
параметр не может быть налл, т.к. по умолчанию required=true

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+//    @GetMapping
+//    public ResponseEntity findAllByFacultyId(@RequestParam Long facultyId) {
```
здесь надо принимать id студента и возвращать факультет

dmitry-bizin replied:
ResponseEntity надо использовать с дженерик-параметром

dmitry-bizin commented on src/main/java/ru/hogwarts/school/model/Faculty.java:
```diff
@@ -0,0 +1,71 @@
+    private Long id;
+    private String name;
+    private String color;
+    @OneToMany(mappedBy = "faculty")
```
поставь здесь JsonIgnore аннотацию перед полем студентов, чтобы не было ошибки

dmitry-bizin commented on src/main/java/ru/hogwarts/school/repositories/StudentRepository.java:
```diff
@@ -0,0 +1,22 @@
+
+    Student findByName(String name);
+
+    Collection <Student> findByAgeBetween(Long age_min, Long age_max); // Добавить эндпоинт для получения всех студентов,
```
названия параметров в camelCase

dmitry-bizin commented on src/main/java/ru/hogwarts/school/repositories/FacultyRepository.java:
```diff
@@ -0,0 +1,20 @@
+public interface FacultyRepository extends JpaRepository<Faculty, Long> {
+
+
+    Collection<Faculty> findByNameOrColorIgnoreCase (String name, String color);
```
для названия не игнорируется регистр, правильнее будет: `findByNameIgnoreCaseOrColorIgnoreCase`

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/StudentServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+        return studentRepository.findByAgeBetween(age_min, age_max);
+    }
+
+//    public Collection<Student> findAllByFacultyId(Long facultyId) {
```
здесь надо принимать id студента, вытаскивать из базы студента и через геттер получать у него факультет

BelousEV committed:
ДЗ 3.4 поправила ещё)

не поняла, как объединить запрос в репозитории по двум параметрам одной переменной
findByNameIgnoreCaseContainingOrColorIgnoreCaseContaining(String nameOrColor);
по-этому сделала 2 метода в репозитории, но в спринге вызывается одним методом с одним параметром

остальное, кажется, сделала

dmitry-bizin commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -65,19 +66,25 @@ public ResponseEntity<Collection<Student>> findStudents(@PathVariable Long age)
-//    }
+    @GetMapping("/getFacultyFromStudentById")
+    public ResponseEntity<Faculty> findFacultyFromStudentById(@RequestParam Long id) {
+        Optional<Student> studentOptional = studentService.findStudent(id);
```
вот это всё можно в метод вынести

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/FacultyServiceImpl.java:
```diff
@@ -44,9 +41,18 @@ public Collection<Faculty> findByColor(String color) {
-    public Collection <Faculty> findByNameOrColorIgnoreCase (String name,String color) { //для БД
-        return facultyRepository.findByNameOrColorIgnoreCase(name, color);
+    public Collection<Faculty> findByNameContainingOrColorContaining(String nameOrColor) { //для БД
+        Collection<Faculty> answer = facultyRepository.findByNameIgnoreCaseContaining(nameOrColor);
```
в репозитории надо написать метод, который принимает 2 параметра и называется `findByNameIgnoreCaseContainingOrColorIgnoreCaseContaining(String name, String color)` и передать в него этот параметр String nameOrColor 2 раза

dmitry-bizin replied:
будет 1 строка в методе

BelousEV committed:
3.5  почти

BelousEV committed:
3.5  кажется, сделала

BelousEV committed:
3.6

BelousEV committed:
3.6

BelousEV committed:
3.6

BelousEV committed:
3.6

BelousEV committed:
3.6

BelousEV committed:
3.6

BelousEV committed:
3.6 rewrite, all tests correct

BelousEV committed:
3.6 mok

BelousEV committed:
3.6 mok

BelousEV committed:
3.6 mok

BelousEV committed:
4.1 and 4.2

BelousEV committed:
4.2

BelousEV committed:
4.2

BelousEV committed:
4.2

BelousEV committed:
4.3 почти готово

BelousEV committed:
4.3 готово

BelousEV committed:
4.3 готово

BelousEV committed:
4.3 готово

BelousEV committed:
4.3 готово

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -1 +1,14 @@
+path.to.avatars.folder = avatars
+
+
+spring.liquibase.change-log=classpath:liquibase/changelog-master.yml
```
здесь неверный путь. У тебя файл лежит в папке scripts, а не в папке liquibase и при старте будет ошибка

BelousEV replied:
файл лежит в папке liquibase, при старте ошибок нет...

dmitry-bizin replied:
может быть у тебя локально так, а на гитхабе вот
![image](https://user-images.githubusercontent.com/15345076/208705742-9c84d8a9-eee3-4c5b-a0de-528297a2ea41.png)

BelousEV committed:
4.4 готово

BelousEV committed:
4.5 готово

BelousEV committed:
4.5 готово

BelousEV committed:
4.5 готово

BelousEV committed:
4.6 готово
